packages/blocks/google-cloud/test/get-recommendations-cli-action.test.ts

```diff
@@ -0,0 +1,209 @@
+const openOpsMock = {
+  ...jest.requireActual('@openops/common'),
+  getUseHostSessionProperty: jest.fn().mockReturnValue({
+    type: 'DYNAMIC',
+    required: true,
+  }),
+  handleCliError: jest.fn(),
+};
+
+jest.mock('@openops/common', () => openOpsMock);
+
+const gcloudCliMock = {
+  runCommand: jest.fn(),
+  runCommands: jest.fn(),
+};
+
+jest.mock('../src/lib/google-cloud-cli', () => gcloudCliMock);
+
+import { getRecommendationsAction } from '../src/lib/actions/get-recommendations-cli-action';
+
+const auth = {
+  clientEmail: 'some-client-email',
+  privateKey: 'some-private-key',
+  projectId: 'some-project-id',
+};
+
+describe('getRecommendationsAction', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should create action with expected props', () => {
+    expect(Object.keys(getRecommendationsAction.props)).toContain(
+      'useHostSession',
+    );
+    expect(getRecommendationsAction.props).toHaveProperty('filterBySelection');
+    expect(getRecommendationsAction.props).toHaveProperty('filterByProperty');
+    expect(getRecommendationsAction.props).toHaveProperty('recommenders');
+    expect(getRecommendationsAction.props).toHaveProperty('location');
+  });
+
+  test('should throw error if recommenders are not set', async () => {
+    const context = createContext({
+      filterByProperty: { project: 'abc' },
+      location: 'europe-west1',
+    });
+
+    await getRecommendationsAction.run(context);
+
+    expect(openOpsMock.handleCliError).toHaveBeenCalledWith({
+      provider: 'Google Cloud',
+      command: '',
+      error: new Error('Recommenders are required'),
+    });
+  });
+
+  test('should return empty property when unauthenticated', async () => {
+    const context = createContext({});
+
+    const result = await getRecommendationsAction.props[
+      'filterByProperty'
+    ].props(
+      {
+        auth: undefined as any,
+        useHostSession: { useHostSessionCheckbox: false },
+      },
+      context,
+    );
+
+    expect(result['markdown']).toMatchObject({
+      type: 'MARKDOWN',
+      description: 'Please authenticate to see the filter options.',
+    });
+  });
+
+  test('should return markdown property if no filterBySelection is selected', async () => {
+    const context = createContext({});
+
+    const result = await getRecommendationsAction.props[
+      'filterByProperty'
+    ].props(
+      {
+        auth,
+        useHostSession: { useHostSessionCheckbox: true },
+        filterBySelection: undefined as any,
+      },
+      context,
+    );
+
+    expect(result).toMatchObject({
+      markdown: {
+        type: 'MARKDOWN',
+        description:
+          'Please select a filter from the dropdown to see the options.',
+      },
+    });
+  });
+
+  test('should return dynamic dropdown property for billingAccount', async () => {
```

Comment:
what about the others?
can maybe do test cases here to not repeat the test code

Reply:
thanks for spotting, i had seperate test cases but i guess i lost them during the rewrite ;(